pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun_test.go

```diff
@@ -0,0 +1,626 @@
+	"github.com/google/go-containerregistry/pkg/registry"
+	v1types "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sigstore/cosign/pkg/cosign"
+	cosignsignature "github.com/sigstore/cosign/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	faketekton "github.com/tektoncd/pipeline/pkg/client/clientset/versioned/fake"
+	remotetest "github.com/tektoncd/pipeline/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakek8s "k8s.io/client-go/kubernetes/fake"
+	"knative.dev/pkg/system"
+	"knative.dev/pkg/webhook/json"
+)
+
+const (
+	nameSpace      = "trusted-task"
+	serviceAccount = "tekton-verify-task-webhook"
+	password       = "hello"
+)
+
+var (
+	// tasks for testing
+	taskSpecTest = &v1beta1.TaskSpec{
+		Steps: []v1beta1.Step{{
+			Container: corev1.Container{
+				Image: "ubuntu",
+				Name:  "echo",
+			},
+		}},
+	}
+	taskSpecTestTampered = &v1beta1.TaskSpec{
+		Steps: []v1beta1.Step{{
+			Container: corev1.Container{
+				Image: "ubuntu",
+				Name:  "hello",
+			},
+		}},
+	}
+
+	ts = &v1beta1.Task{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "tekton.dev/v1beta1",
+			Kind:       "Task"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-task",
+			Namespace: nameSpace,
+		},
+		Spec: *taskSpecTest,
+	}
+	tsTampered = &v1beta1.Task{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "tekton.dev/v1beta1",
+			Kind:       "Task"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-task-tampered",
+			Namespace: nameSpace,
+		},
+		Spec: *taskSpecTestTampered,
+	}
+
+	trTypeMeta = metav1.TypeMeta{
+		Kind:       pipeline.TaskRunControllerName,
+		APIVersion: "tekton.dev/v1beta1"}
+
+	trObjectMeta = metav1.ObjectMeta{
+		Name:        "tr",
+		Namespace:   nameSpace,
+		Annotations: map[string]string{},
+	}
+
+	// service account
+	sa = &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: nameSpace,
+			Name:      serviceAccount,
+		},
+	}
+
+	k8sclient    = fakek8s.NewSimpleClientset(sa)
+	tektonClient = faketekton.NewSimpleClientset(ts, tsTampered)
+)
+
+func init() {
+	os.Setenv("SYSTEM_NAMESPACE", nameSpace)
+	os.Setenv("WEBHOOK_SERVICEACCOUNT_NAME", serviceAccount)
+}
+
+func TestVerifyResources_TaskSpec(t *testing.T) {
+	ctx := context.Background()
+
+	// Get singer
+	signer, err := getSingerFromFile(t, ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tr := v1beta1.TaskRun{
+		TypeMeta:   trTypeMeta,
+		ObjectMeta: trObjectMeta,
+		Spec: v1beta1.TaskRunSpec{
+			TaskSpec: &ts.Spec,
+		},
+	}
+
+	apiTaskRun := TrustedTaskRun{TaskRun: tr}
+	taskRunPass := apiTaskRun.DeepCopy()
+	taskRunContentTampered := apiTaskRun.DeepCopy()
+	taskRunUnsigned := apiTaskRun.DeepCopy()
+
+	// Sign task
+	trList := []*TrustedTaskRun{
+		taskRunPass,
+		taskRunContentTampered,
+	}
+
+	for _, tr := range trList {
```

Comment:
This is only used once and isn't that long, so feel free to inline this - 
```suggestion
	for _, tr := range []*TrustedTaskRun{taskRunPass, taskRunContentTampered} {
```